tools/gulp/tasks/release.ts

```diff
@@ -1,50 +1,106 @@
 import {spawn} from 'child_process';
-import {existsSync, statSync} from 'fs';
-import {task} from 'gulp';
-import gulpRunSequence = require('run-sequence');
-import path = require('path');
-import minimist = require('minimist');
-
-import {execTask, cleanTask} from '../util/task_helpers';
-import {DIST_COMPONENTS_ROOT} from '../constants';
-
+import {existsSync, statSync, copySync, writeFileSync} from 'fs-extra';
+import {join} from 'path';
+import {task, src, dest} from 'gulp';
+import {execTask, sequenceTask} from '../util/task_helpers';
+import {
+  DIST_RELEASE, DIST_BUNDLES, DIST_MATERIAL, COMPONENTS_DIR, LICENSE_BANNER
+} from '../constants';
+import * as minimist from 'minimist';
+
+// There are no type definitions available for these imports.
+const gulpRename = require('gulp-rename');
+
+/** Parse command-line arguments for release task. */
 const argv = minimist(process.argv.slice(3));
 
-/** Removes redundant spec files from the release. TypeScript creates definition files for specs. */
-// TODO(devversion): tsconfig files should share code and don't generate spec files for releases.
-task(':build:release:clean-spec', cleanTask('dist/**/*+(-|.)spec.*'));
+// Matches all Typescript definition files for Material.
+const typingsGlob = join(DIST_MATERIAL, '**/*.d.ts');
+// Matches the "package.json" and "README.md" file that needs to be shipped.
+const assetsGlob = join(COMPONENTS_DIR, '+(package.json|README.md)');
+// Matches all UMD bundles inside of the bundles distribution.
+const umdGlob = join(DIST_BUNDLES, '*.umd.*');
+// Matches all flat ESM bundles (e.g material.js and material.es5.js)
+const fesmGlob = [join(DIST_BUNDLES, '*.js'), `!${umdGlob}`];
+
+task('build:release', sequenceTask(
+  'library:build',
+  ':package:release',
+));
+
+/** Task that combines intermediate build artifacts into the release package structure. */
+task(':package:release', [
+  ':package:fix-metadata',
+  ':package:metadata',
```

Comment:
Should only call `:packagae:metadata`, which then depends on `:package:fix-metadata`?
(to make sure the order of operations is correct since this isn't a `runSequence`)